File: backend/app/schemas/cashbook.py
"""
Cashbook schemas
"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class CashbookEntryResponse(BaseModel):
    id: int
    business_id: int
    entry_type: str
    amount: float
    payment_method: Optional[str] = None
    reference_id: Optional[int] = None
    reference_type: Optional[str] = None
    description: str
    entry_date: datetime
    created_by: Optional[int] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


class CashReconciliationCreate(BaseModel):
    reconciliation_date: Optional[datetime] = None
    actual_cash: float
    adjustment_reason: Optional[str] = None
    notes: Optional[str] = None


class CashReconciliationResponse(BaseModel):
    id: int
    business_id: int
    reconciliation_date: datetime
    expected_cash: float
    actual_cash: float
    difference: float
    adjustment_amount: float
    adjustment_reason: Optional[str] = None
    notes: Optional[str] = None
    reconciled_by: Optional[int] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


class CashbookSummary(BaseModel):
    date: str
    cash_in: float
    cash_out: float
    net_cash: float
    cash_by_method: dict  # {"cash": 100, "telebirr": 50, "bank": 200}
    entries: List["CashbookEntryResponse"]
    
    class Config:
        from_attributes = True

File: frontend/lib/api/cashbook.ts
import { apiClient } from './client'

export interface CashbookEntry {
  id: number
  business_id: number
  entry_type: string // payment_in, expense_out, adjustment
  amount: number
  payment_method?: string
  reference_id?: number
  reference_type?: string
  description: string
  entry_date: string
  created_by?: number
  created_at: string
}

export interface CashbookSummary {
  date: string
  cash_in: number
  cash_out: number
  net_cash: number
  cash_by_method: {
    cash: number
    telebirr: number
    bank: number
    other: number
  }
  entries: CashbookEntry[]
}

export interface CashReconciliation {
  id: number
  business_id: number
  reconciliation_date: string
  expected_cash: number
  actual_cash: number
  difference: number
  adjustment_amount: number
  adjustment_reason?: string
  notes?: string
  reconciled_by?: number
  created_at: string
}

export interface CashReconciliationCreate {
  reconciliation_date?: string
  actual_cash: number
  adjustment_reason?: string
  notes?: string
}

export const cashbookApi = {
  getSummary: async (date?: string, period: 'day' | 'week' | 'month' = 'day'): Promise<CashbookSummary> => {
    const params: any = { period }
    if (date) params.date = date
    const response = await apiClient.get('/cashbook/summary', { params })
    return response.data
  },

  reconcile: async (data: CashReconciliationCreate): Promise<CashReconciliation> => {
    const response = await apiClient.post('/cashbook/reconcile', data)
    return response.data
  },
}

